### DISCUSSION TITLE
Support Terragrunt using .git extension in Terreform source path

### DISCUSSION THREAD

**Discussion by cxn-sre-terraform** [renovatebot/renovate#28728] [Suggest an Idea]
### Tell us more.

# Use case

Currently we use renovatebot to keep our terragrunt infrastructure up-to-date. Unfortunately, we faced and issue. The `packageName` built in a wrong format and it's not able to find the module  itself to update.

We use full path of the module with `.git` extension. Like this:

```terraform
source = "git::https://gitlab.com/path/to/modules/test.git//.?ref=1.0.12345678901234"
```

# Issue description

Renovate generates `path/to/modules/test.git` `packageName` as Gitlab repository path which obviously cannot be found because of the `.git` part of it.

```bash
"Failed to look up gitlab-tags package path/to/modules/test.git",
```

With the following terraform source it works:

```terraform
source = "git::https://gitlab.com/path/to/modules/test//.?ref=1.0.12345678901234"
```

# Workaround

We implemented a temporary workaround as well. We used a regex to cut the `.git` part of the `source` and with this it's able to bump the version without any issue.

```json
  "regexManagers": [
        {
      "fileMatch": [
        "terragrunt.hcl$",
        "terragrunt.hcl.j2$"
      ],
      "matchStrings": [
        "(?:git::)?(?<url>(?:http|https|ssh):\/\/(?:.*@)?(?<host>gitlab.com[^/]*)\/(?<path>.*))[.].*git.*\?(depth=\d+&)?ref=(?<currentValue>.+)\""
      ],
      "packageNameTemplate": "{{path}}",
      "datasourceTemplate": "gitlab-tags",
      "depNameTemplate": "{{host}}/{{path}}"
    }
  ]
```

# Summary

Would you be so kind to help in this case of ours? Is it a bug or an intended change? Terragrunt documentation contains `source` with the `.git` path -> https://terragrunt.gruntwork.io/docs/features/keep-your-terraform-code-dry/#using-terragrunt-with-private-git-repos.

Thanks in advance.

**Comment by rarkins** [comment]
You're welcome to submit a PR making this change permanent within the terragrunt manager

**Comment by viceice** [comment]
I think this was fixed in
- https://github.com/renovatebot/renovate/pull/28739
- https://github.com/renovatebot/renovate/releases/tag/37.329.0

**Reply by cxn-sre-terraform** [reply]
Yup that release contains the fix what was I need. Thanks for your help!
